python/podio/podio_config_reader.py

```diff
@@ -85,18 +85,26 @@ def _bare_member_conv(result):
 
   def parse(self, string, require_description=True):
     """Parse the passed string"""
-    matchers_cbs = [
+    default_matchers_cbs = [
         (self.full_array_re, self._full_array_conv),
-        (self.member_re, self._full_member_conv)
-        ]
-
-    if not require_description:
-      matchers_cbs.extend((
-          (self.bare_array_re, self._bare_array_conv),
-          (self.bare_member_re, self._bare_member_conv)
-          ))
-
-    return self._parse_with_regexps(string, matchers_cbs)
+        (self.member_re, self._full_member_conv)]
+
+    no_desc_matchers_cbs = [
+        (self.bare_array_re, self._bare_array_conv),
+        (self.bare_member_re, self._bare_member_conv)]
+
+    if require_description:
+      try:
+        return self._parse_with_regexps(string, default_matchers_cbs)
+      except DefinitionError:
+        # check whether we could parse this if we don't require a description and
+        # provide more details in the error if we can
+        self._parse_with_regexps(string, no_desc_matchers_cbs)
+        # pylint: disable-next=raise-missing-from
+        raise DefinitionError(f"'{string}' is not a valid member definition. Description comment is missing.\n"
+                               + "Correct Syntax: <type> <name> // <comment>")
```

Comment:
```suggestion
        raise DefinitionError(f"'{string}' is not a valid member definition. Description comment is missing."
                               " Correct Syntax: <type> <name> // <comment>")
```
You don't need the `+` in this case, and I would also not put in a newline (`\n`) to the output. Other than that this looks good.

Reply:
I put the newline character so that the error message looks good.
With `newline character`
![image](https://github.com/AIDASoft/podio/assets/90386813/a6d61df3-b1ec-4d06-836f-412b358bb602)
Without `newline character`
![image](https://github.com/AIDASoft/podio/assets/90386813/405d241c-d3d4-4519-bf77-777ddb3b21b1)
Without `newline character` the error doesn't look good on screen.

Reply:
OK. I like the newline. But the `+` is not necessary for string concatenation in this case, because it will be done automatically inside the `()`.

Reply:
Okay, I'll keep the newline character and remove + .